src/realm/sync/noinst/protocol_codec.hpp

```diff
@@ -395,11 +406,26 @@ class ClientProtocol {
         progress.latest_server_version.salt = msg.read_next<salt_type>();
         progress.upload.client_version = msg.read_next<version_type>();
         progress.upload.last_integrated_server_version = msg.read_next<version_type>();
-        auto query_version = connection.is_flx_sync_connection() ? msg.read_next<int64_t>() : 0;
 
-        // If this is a PBS connection, then every download message is its own complete batch.
-        auto last_in_batch = connection.is_flx_sync_connection() ? msg.read_next<bool>() : true;
-        auto downloadable_bytes = msg.read_next<int64_t>();
+        DownloadInfoExtra info;
+        if (is_flx) {
+            info.query_version = msg.read_next<int64_t>();
```

Comment:
we could update `peek_token_impl()` to handle doubles and the code below would be simplified.

Reply:
moved the implementation to HeaderLineParser